libtiledbsoma/src/soma/soma_dataframe.cc

```diff
@@ -180,17 +182,78 @@ void SOMADataFrame::update_dataframe_schema(
                 attr_type,
                 enmr_type,
                 ordered));
-            se.add_enumeration(Enumeration::create_empty(
-                *ctx->tiledb_ctx(),
-                attr_name,
-                ArrowAdapter::to_tiledb_format(enmr_type),
-                enmr_type == "u" || enmr_type == "z" || enmr_type == "U" ||
-                        enmr_type == "Z" ?
-                    TILEDB_VAR_NUM :
-                    1,
-                ordered));
-            AttributeExperimental::set_enumeration_name(
-                *ctx->tiledb_ctx(), attr, attr_name);
+
+            // First we need to check if the array has (ever had) an enumeration
+            // of this type in its history.
+            //
+            // Note that in core, attrs have names and enumerations have
+            // names, and the names need not match -- and multiple attrs can
+            // use the same enumeration. For tiledbsoma, though, we don't take
+            // advantage of this flexibility: we use the same name for an attr
+            // and its enumeration (if if has one).
+            //
+            // It's quite possible the array had this name as an enumerated
+            // column, and then that column was dropped, and it's now being
+            // re-added.
+            //
+            // * We need to check if this is the case.
+            // * If it is, we need to check if the enumeration is compatible.
+            try {
+                const auto existing_enmr = ArrayExperimental::get_enumeration(
+                    tctx, array, attr_name);
```

Comment:
Are we able to use this instead of opening a new `Array`?

```suggestion
                const auto existing_enmr = ArrayExperimental::get_enumeration(
                    tctx, *arr_, attr_name);
```

Reply:
@nguyenv this is a static method